mayastor/src/bin/mayastor-client/context.rs

```diff
@@ -1,7 +1,35 @@
 use crate::{BdevClient, JsonClient, MayaClient};
 use byte_unit::Byte;
+use bytes::Bytes;
 use clap::ArgMatches;
-use std::cmp::max;
+use http::uri::{Scheme, Uri, Authority, PathAndQuery};
+use snafu::{Backtrace, ResultExt, Snafu};
+use std::{cmp::max, str::FromStr};
+use tonic::transport::Endpoint;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Invalid URI bytes: {}", source))]
```

Comment:
You don't need to add the source with snafu

Reply:
Apologies I misremembered this! You do have the error on the chain and so if you loop the chain and print it then you get all of the errors, but if you just use the Display/tostring then you do not get the source printed out. On the other hand, if you add the source to the display but want the full chain and walk it, then you end up with sources printed out twice :(